site/static/diagrams/gameserver-reserved.puml

```diff
@@ -0,0 +1,74 @@
+@startuml
+participant Matchmaker
+participant Agones
+participant "Game Server\nProcess" as Binary
+participant "GameServer\nResource" as GameServer
+participant SDK
+box "Game Server Pod"
+  participant Binary
+  participant SDK
+end box
+
+== GameServer Start ==
+
+Agones -> GameServer: GameServer created through\na <i>Fleet</i> configuration
+activate GameServer
+GameServer -> Binary: Agones creates a Pod with the\nconfigured Game Server Container
+activate Binary
+activate SDK
+Binary -> SDK: SDK.Health()
+note right
+  <i>Health()</i> is a continuous
+  ping that occurs under
+  the configured threshold.
+end note
+GameServer <-- SDK: Maintains Healthy status
+
+== Match Maker Registration ==
+group loop
+Binary -> SDK: SDK.Reserve(duration)
+GameServer <-- SDK: Set state to <i>Reserved</i> for duration
+note right
+  <i>duration</i> should be longer
+  than how long the matchmaker
+  requires <i>Register()</i> to be
+  available for, so it will not be
+  scaled down
+end note
+Binary -> Matchmaker: Register()
+note right
+  Register with the Matchmaker as
+  available to host a game session
+end note
+note left
+match maker registration
```

Comment:
Is match maker one work (e.g. matchmaker)?

Reply:
Yep, good catch.